public/project/javascripts/directives/ngHover.js

```diff
@@ -0,0 +1,28 @@
+app.directive('ngHover', function($rootScope, StateService) {
+  return {
+    link: function(scope, element, attrs) {
+      element.bind('mouseover', function(event) {
+
+        var obj = {};
+
+        var isHomeRow = (event.target.id === 'home-row') || (event.target.id === 'home-row-copy');
+        var isCarsRow = (event.target.id === 'cars-row') || (event.target.id === 'cars-row-copy');
+        var isOutsideSidebar = (event.target.parentNode.id === 'landing-view') || (event.target.parentNode.id === 'nav-view');
+
+        if (!isOutsideSidebar) {
+          scope.hoverHome = isHomeRow;
+          scope.hoverCars = isCarsRow;
+          scope.$apply();
+
+          /* only broadcast if the sidebar is both toggled and fully open */
+        } else if (StateService.data['sidebar'].isToggled && StateService.data['sidebar'].isFullyOpen) {
+          obj.container = 'sidebar';
+          obj.key = 'isToggled';
+          $rootScope.$broadcast('toggleSidebar', obj);
+          $rootScope.$apply();
+        }
+      });
+    }
+  }
+  ngHover.$inject('$rootScope, StateService');
+});
```

Comment:
This really gooood. But breggh, there is a ng-mouseenter and ng-mouseleave built into angular. For the hover effects, you can just use :hover in the css. But this is a well written directive